File: backend/src/main/java/com/platform/entity/Module.java
package com.platform.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("sys_module")
public class Module {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String moduleKey;

    private String name;

    private String version;

    private String description;

    private String author;

    private String icon;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/api/module.ts
import request from '@/utils/request'

export interface Module {
  id: number
  moduleKey: string
  name: string
  version: string
  description?: string
  author?: string
  icon?: string
  status: number
  createdAt?: string
  updatedAt?: string
}

export const getModuleList = () => {
  return request.get<Module[]>('/v1/modules')
}

export const getModule = (id: number) => {
  return request.get<Module>(`/v1/modules/${id}`)
}

export const toggleModule = (id: number) => {
  return request.put(`/v1/modules/${id}/toggle`)
}
